runtime/src/main/java/org/corfudb/runtime/collections/TimingHistogram.java

```diff
@@ -0,0 +1,91 @@
+package org.corfudb.runtime.collections;
+
+import com.google.gson.JsonObject;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.distribution.HistogramSnapshot;
+import org.corfudb.common.metrics.micrometer.MeterRegistryProvider;
+
+import javax.management.timer.Timer;
+import java.util.concurrent.TimeUnit;
+import java.util.Optional;
+
+/**
+ * Minimalistic stats to capture time taken.
+ *
+ * created by hisundar 2020-09-20
+ */
+public class TimingHistogram {
+    private final Optional<DistributionSummary> timer;
+
+    public TimingHistogram(String metricName, String tag) {
+        this.timer = MeterRegistryProvider.getInstance().map(registry ->
+                DistributionSummary
+                        .builder(metricName)
+                        .tag("tableName", tag)
+                        .publishPercentiles(0.50, 0.95, 0.99)
+                        .publishPercentileHistogram()
+                        .baseUnit("s")
+                        .register(registry));
+    }
+
+    /**
+     * Record the time elapsed. Recompute stats.
+     *
+     * @param val - time elapsed in nano seconds
+     */
+    public void update(long val) {
+        if (!timer.isPresent()) {
+            return;
+        }
+        timer.get().record(val);
+    }
+
+    /**
+     * Convert elapsed time in nanoseconds into string with human readable suffix.
+     *
+     * @param nanos - avg, max or min to suffix
+     * @return the stat param with the coarsest time suffix
+     */
+    public static String coarsestGranularString(double nanos) {
```

Comment:
how it could be possible fractional nanoseconds?
I'd like to propose using long, you convert it in long either way